### PR TITLE
Machine Translation Docs

### PR DESCRIPTION
**Pull request by MaximumEntropy**

### CODE REVIEW

File: docs/source/nlp/machine_translation.rst [hunk 1]
```diff
@@ -0,0 +1,251 @@
+.. _machine_translation:
+
+Machine Translation Models
+=========================
```
Comment (ericharper) [inline docs/source/nlp/machine_translation.rst:4]:
```suggestion
==========================
```

File: docs/source/nlp/machine_translation.rst [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
++-------------------------------------------------------------+-----------------+----------------+------------------------------------------------------------------------------------------------+
+| model.{train_ds,validation_ds,test_ds}.tokens_in_batch      | int             | 512            | Maximum number of tokens per minibatch                                                         |
++-------------------------------------------------------------+-----------------+----------------+------------------------------------------------------------------------------------------------+
+| model.{train_ds,validation_ds,test_ds}.clean                | bool            | true           | Whether to clean the dataset by discarding examples that are greater than max_seq_length       |
++-------------------------------------------------------------+-----------------+----------------+------------------------------------------------------------------------------------------------+
+| model.{train_ds,validation_ds,test_ds}.max_seq_length       | int             | 512            | Path to pre-computed vocab file if exists                                                      |
```
Comment (ericharper) [inline docs/source/nlp/machine_translation.rst:222]: Description is for vocab_file should be for max_seq_length

File: docs/source/nlp/machine_translation.rst [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
++-------------------------------------------------------------+-----------------+----------------+------------------------------------------------------------------------------------------------+
+| model.{train_ds,validation_ds,test_ds}.cache_ids            | bool            | true           | Whether to cache IDs to avoid re-tokenizing data                                               |
++-------------------------------------------------------------+-----------------+----------------+------------------------------------------------------------------------------------------------+
+| model.{train_ds,validation_ds,test_ds}.cache_data_per_node  | bool            | false          | Whether to cache IDs in each of the nodes in multi-node training                               |
++-------------------------------------------------------------+-----------------+----------------+------------------------------------------------------------------------------------------------+
+| model.{train_ds,validation_ds,test_ds}.use_cache            | bool            | true           | Whether to use the cache if available                                                          |
```
Comment (ericharper) [inline docs/source/nlp/machine_translation.rst:228]: Let's mention that these are being deprecated and to use tarred dataset for caching.

Comment (MaximumEntropy) [reply docs/source/nlp/machine_translation.rst:228]: Sure, would you still suggest that I include them in this table?

Comment (okuchaiev) [reply docs/source/nlp/machine_translation.rst:228]: let's drop deprecated stuff from docs

**Comment by okuchaiev** [comment]
looks good to me. Please remove "Draft" status when you are done.

File: docs/source/nlp/machine_translation.rst (hunk 2 above)
Comment (MaximumEntropy) [reply docs/source/nlp/machine_translation.rst]: Done. Thanks!

File: docs/source/nlp/machine_translation.rst (hunk 3 above)
Comment (MaximumEntropy) [reply docs/source/nlp/machine_translation.rst]: Cool, done.

**Review by okuchaiev — APPROVED** [review]
lgtm

**Pull request merged**